db.py

```diff
@@ -42,34 +41,56 @@ def addView(self, connection, url):
 
     def getTopSites(self, connection, amount=10):
         """ Get the top domains using this tool by hits. Ignore specified domains """
+        top_urls = self.getTopUrls(connection, -1)
+
+        # Get total hits per domain
+        site_counts = defaultdict(int)
+        for url in top_urls['urls']:
+            # Get the domain - part before the first '/'
+            domain = url.split('/')[0]
+            # Add hit counts to the domain
+            site_counts[domain] += top_urls['values'][url]
+
+        # Sort the domains by hits
+        sorted_sites = sorted(site_counts, key=lambda x: site_counts[x], reverse=True)
+
+        # Return sorted domains and their values, this allows for lower Python version support
+        return {
+            'domains': sorted_sites[:amount],
+            'values': {site: site_counts[site] for site in site_counts}
+        }
+
+    def getTopUrls(self, connection, amount=10):
+        """ Get the top urls using this tool by hits. Ignore specified domains """
         # Select all urls and counts
+        if amount < 0:
+            amount = 2147483647
         cursor = connection.cursor()
-        cursor.execute('select url, count from url')
+        cursor.execute(f'select url, count from url order by count desc limit {amount + len(config.TOP_SITES_IGNORE_DOMAIN_RE_MATCH)}')
         urls_and_counts = cursor.fetchall()
 
         # Get total hits per domain
-        site_counts = defaultdict(int)
+        url_counts = defaultdict(int)
         for row in urls_and_counts:
-            if row[0] == b'':
+            url = row[0]
+            if url == b'':
                 continue
-            # Get the domain - part before the first '/'
-            domain = row[0].split('/')[0]
-            # Check if domain is on the ignore list
+            # Check if url is on the ignore list
             on_ignore = False
             for regex in config.TOP_SITES_IGNORE_DOMAIN_RE_MATCH:
```

Comment:
The name of `TOP_SITES_IGNORE_DOMAIN_RE_MATCH` hints these regular expressions are for domains. We should either change the name or only use these to check domains.

I would rather this list to only be expressions to check domains because if this is provided with a URL like `www.example.com/post/what-is-192.168.1.1`, it will be filtered incorrectly.